src/items.rs

```diff
@@ -1905,12 +1905,32 @@ fn get_missing_arg_comments(
 
 impl Rewrite for ast::Param {
     fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
+        let param_attrs_result = self
+            .attrs
+            .to_vec()
+            .rewrite(context, Shape::legacy(shape.width, shape.indent))?;
+        let span = if !self.attrs.is_empty() {
+            mk_sp(
+                self.attrs[self.attrs.len() - 1].span.hi(),
+                self.pat.span.lo(),
+            )
+        } else {
+            mk_sp(self.span.lo(), self.span.lo())
+        };
+
         if let Some(ref explicit_self) = self.to_self() {
-            rewrite_explicit_self(context, explicit_self)
+            rewrite_explicit_self(context, explicit_self, &param_attrs_result, span, shape)
         } else if is_named_arg(self) {
-            let mut result = self
-                .pat
-                .rewrite(context, Shape::legacy(shape.width, shape.indent))?;
+            let mut result = combine_strs_with_missing_comments(
+                context,
+                &param_attrs_result,
+                &self
+                    .pat
+                    .rewrite(context, Shape::legacy(shape.width, shape.indent))?,
+                span,
+                shape,
+                true,
```

Comment:
Am I correct in assuming that this (`allow_extend`) should always be true? If not, what are the conditions used to determine the value?

Reply:
In this case we can always set this to `true`. In other place, we may want to change the value depending on the configuration.

Reply:
Actually, we should set this `false` if the attributes consists of multiple lines:

```rust
// ok
fn foo(#[foo] param: u32) {
    // ...
}

// bad
fn foo(
    #[foo]
    #[bar] param: u32,
) {
    // ...
}

// ok
fn foo
    #[foo]
    #[bar]
    param: u32,
) {
    // ...
}
```

Reply:
I personally agree with this too, though in the tests added in the PR to support param attributes,  https://github.com/rust-lang/rust/pull/60669/files?file-filters%5B%5D=.rs#diff-65a6151def3a5d7f3a4191644483f941R39-R40, it seems like they are still doing same line for the last attribute and the param 🤷‍♂

Reply:
I went ahead and made this update. If later on folks _really_ have a need/desire to be able to put the last attribute and param def on the same line, then we cross that bridge at that time (perhaps with a config option)